.github/workflows/ci.yaml

```diff
@@ -0,0 +1,46 @@
+---
+name: DNF CI
+on: pull_request_target
+
+jobs:
+  integration-tests:
+    name: Integration Tests
+    runs-on: ubuntu-latest
+    container:
+      image: fedora:latest
+      options: --privileged
+    steps:
+      - name: Install API token for Copr
+        id: copr
+        run: |
+          # hack: Github replaces secrets with *** in the whole output (even in
+          # e.g. Copr URLs printed by rpm-gitoverlay). If there's a comment (#)
+          # at the end of the secret (e.g.  "rpmsofwaremanagement #"), this
+          # will clean it up and since it is no longer the whole secret being
+          # printed, Github won't hide it anymore.
+          COPR_USER=${{secrets.COPR_USER}}
+          if [ -z "$COPR_USER" ]; then echo "COPR_USER secret is required to run the CI."; exit 1; fi
+          echo "::set-output name=copr-user::$COPR_USER"
+
+          mkdir -p "$HOME/.config"
+          echo "${{secrets.COPR_API_TOKEN}}" > "$HOME/.config/copr"
+
+      - name: Check out ci-dnf-stack
+        uses: actions/checkout@v2
+        with:
+          repository: rpm-software-management/ci-dnf-stack
+
+      - name: Setup CI
+        uses: ./.github/actions/setup-ci
+
+      - name: Check out sources
+        uses: actions/checkout@v2
+        with:
+          path: gits/${{github.event.pull_request.head.repo.name}}
+          ref: ${{github.event.pull_request.head.sha}}  # check out the PR HEAD
+          fetch-depth: 0
+
+      - name: Run CI
+        uses: ./.github/actions/run-ci
+        with:
+          copr-user: ${{steps.copr.outputs.copr-user}}
```

Comment:
Why don't you use `copr-usr: ${{ secrets.COPR_USER }}` here directly, and drop the weird hack above? The *** replacement only happens in the output, not the actual value substitution.

Reply:
I assume by "weird hack" (I do agree with the choice of words :slightly_smiling_face:) you mean `COPR_USER=${{secrets.COPR_USER}}`. I need this, without it Github will scrub the output of all occurrences of the value of `secrets.COPR_USER`, which end up in URLs to the Copr builds that are printed druing the build process.

I could put it into the `run-ci` action to share a bit more code. I did it this way as a I also took the opportunity to exit early if `COPR_USER` isn't set (on line 22). If I don't exit early, it will call the `setup-ci` action which installs some stuff and takes a couple of minutes, then bail out...

Reply:
Ah, so you depend on URLs inside the workflow log? I see. Fine then :-)

Reply:
I don't hard-depend on them, but they are there for viewer's convenience, this would render them (mostly) useless...